### PR TITLE
Fix creative mode screen to reference CreativeModeTab::getTabsImage (#9627)

### LINKED ISSUE #9627: CreativeModeTab's .getTabsImage() method isn't used anywhere. (Cannot set custom texture for creative menu tabs)
**Minecraft Version:** 1.20.1

**Forge Version:** 47.0.50

**Steps to Reproduce:**
 1. Register a CreativeModeTab and use .withTabsImage() in builder with some custom texture.
 2. Open inventory and see that the texture of the tabs haven't changed.

**Description of issue:** .getTabsImage() method from CreativeModeTab isn't used anywhere which leads to a problem where you can't set a custom texture for tabs in creative menu.

### PR DESCRIPTION
**Pull request by nathanatnorthernbingedotcom**
The *CreativeModeTab.Builder* class exposes two methods for getting and setting the background image of tabs in the creative mode inventory; however, the background image was not subsequently used. This commit patches *CreativeModeInventoryScreen* to query the background image from each *CreativeModeTab*, instead of using a constant.

This allows modders to use existing forge methods to define custom background textures for their creative mode tabs. (Fixes #9627)

An example was added to the existing *CreativeModeTabTest*.

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java.patch [hunk 1]
```diff
@@ -1,5 +1,13 @@
 --- a/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java
 +++ b/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java
+@@ -49,7 +_,6 @@
+ 
+ @OnlyIn(Dist.CLIENT)
+ public class CreativeModeInventoryScreen extends EffectRenderingInventoryScreen<CreativeModeInventoryScreen.ItemPickerMenu> {
+-   private static final ResourceLocation f_98504_ = new ResourceLocation("textures/gui/container/creative_inventory/tabs.png");
```
Comment (Jonathing) [inline patches/minecraft/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java.patch]: Patching this out seems unnecessary, even if it is now unused.

Comment (nathanatnorthernbingedotcom) [reply patches/minecraft/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java.patch]: Reverted

Comment (Jonathing) [reply patches/minecraft/net/minecraft/client/gui/screens/inventory/CreativeModeInventoryScreen.java.patch]: Awesome

**Review by Jonathing — APPROVED** [review]
Absolutely solid. Very concise with minimal changes; you love to see it.

Proof that it works

**Comment by LexManos** [comment]
Pulled in https://github.com/MinecraftForge/MinecraftForge/commit/fc7af96bde3e8b5da64232dde68f181980f69962

**Pull request closed without merging**